src/maintypes.jl

```diff
@@ -36,3 +36,31 @@ matrix internally, it is possible to use [`@manifold_element_forwards`](@ref) an
 [`@default_manifold_fallbacks`](@ref) to reduce implementation overhead.
 """
 abstract type AbstractManifoldPoint end
+
+"""
+    abstract type AbstractManifoldSize end
+
+Abstract representation of manifold size. Can be either [`StaticSize`](@ref) or
+[`RTSize`](@ref).
+"""
+abstract type AbstractManifoldSize end
+
+"""
+    StaticSize{T}
+
+Static size of a manifold.
+"""
+struct StaticSize{T} <: AbstractManifoldSize end
+StaticSize(t::NTuple) = StaticSize{t}()
+
+"""
+    RTSize{TS<:NTuple{N,Int} where N}
+
+Runtime size of a manifold. 
+"""
+struct RTSize{TS<:NTuple{N,Int} where {N}} <: AbstractManifoldSize
```

Comment:
I think `RuntimeSize` is not too bad here.

Though maybe another pair of possible names might be `ParameterSize` and `Fieldsize` or something like that?

Reply:
I think I prefer `RuntimeSize`.

Reply:
Runtime and static are more like meta-implications of this difference in storing the parameter.

I think I'd prefer names that not address the meta level directly?

One is a parameter the other a field and that is the difference. Or phrased differently. I read runtime but I have to think field – I read static but I have to think parameter. This one step in thinking is what I do not like in this naming so much.

Reply:
I see, that's a good point. I will change it to `ParameterSize` and `FieldSize`.

Reply:
Besides the Size name ;) But `ParameterParameter` sounds silly. But `ParameterX` and `FieldX` are good, we just need a better name for `X`.

Reply:
Hm though these parameters are not directly the dimension it is what they store so maybe `X` as `Dimension` would be best, though to be precise it is actually DimensionInformation that is stored?